osu.Game.Rulesets.Rush/Input/RushInputManager.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Shane Woolcock. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using osu.Framework.Extensions.ListExtensions;
+using osu.Framework.Input;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Lists;
+using osu.Game.Rulesets.Rush.Objects;
+using osu.Game.Rulesets.Rush.Replays;
+using osu.Game.Rulesets.UI;
+using osuTK.Input;
+
+namespace osu.Game.Rulesets.Rush.Input
+{
+    public class RushInputManager : RulesetInputManager<RushAction>
+    {
+        protected override bool MapMouseToLatestTouch => false;
+        public new RushFramedReplayInputHandler ReplayInputHandler => (RushFramedReplayInputHandler)base.ReplayInputHandler;
+
+        /// <summary>
+        /// Retrieves all actions in a currenty pressed states.
+        /// </summary>
+        public SlimReadOnlyListWrapper<RushAction> PressedActions => ((List<RushAction>)KeyBindingContainer.PressedActions).AsSlimReadOnly();
+
+        public RushInputManager(RulesetInfo ruleset)
+            : base(ruleset, 0, SimultaneousBindingMode.Unique)
+        {
+        }
+
+        protected override MouseButtonEventManager CreateButtonEventManagerFor(MouseButton button)
+            => new RushMouseEventManager(button, this);
+
+        protected override TouchEventManager CreateButtonEventManagerFor(TouchSource source)
+            => new RushTouchEventManager(source, this);
+
+
+        private readonly Dictionary<TouchSource, RushAction> touchActionMap = new Dictionary<TouchSource, RushAction>();
+
+        private readonly Dictionary<RushAction, bool> actionsInUse = new Dictionary<RushAction, bool>();
+
+        private void updateActionsCache()
+        {
+            for (RushAction action = RushAction.GroundPrimary; action <= RushAction.Fever; ++action)
+                actionsInUse[action] = false;
+
+            foreach (var pressedAction in PressedActions)
+                actionsInUse[pressedAction] = true;
+        }
+
+        private RushAction? tryGetGroundAction() => tryGetActionInRange(RushAction.GroundPrimary, RushAction.AirPrimary);
+        private RushAction? tryGetAirAction() => tryGetActionInRange(RushAction.AirPrimary, RushAction.Fever);
+        private RushAction? tryGetFeverAction()
+        {
+            actionsInUse.TryGetValue(RushAction.Fever, out var inUse);
+
+            if (!inUse) return RushAction.Fever;
+
+            return null;
+        }
+        private RushAction? tryGetActionInRange(RushAction lowerBound, RushAction upperBound)
+        {
+            for (RushAction a = lowerBound; a < upperBound; ++a)
+            {
+                actionsInUse.TryGetValue(a, out var inUse);
+
+                if (!inUse) return a;
+            }
+
+            return null;
+        }
+
+        public void TryPressTouchAction(TouchSource source, TargetAction action)
```

Comment:
Since the name has `Try` in it, please return a `bool` indicating whether it was successful.